contracts/contracts/strategies/crosschain/AbstractCCTPIntegrator.sol

```diff
@@ -0,0 +1,568 @@
+        external
+        onlyGovernor
+    {
+        _setMinFinalityThreshold(_minFinalityThreshold);
+    }
+
+    /**
+     * @dev Set the minimum finality threshold
+     * @param _minFinalityThreshold Minimum finality threshold
+     */
+    function _setMinFinalityThreshold(uint32 _minFinalityThreshold) internal {
+        // 1000 for fast transfer and 2000 for standard transfer
+        require(
+            _minFinalityThreshold == 1000 || _minFinalityThreshold == 2000,
+            "Invalid threshold"
+        );
+
+        minFinalityThreshold = _minFinalityThreshold;
+        emit CCTPMinFinalityThresholdSet(_minFinalityThreshold);
+    }
+
+    /**
+     * @dev Set the fee premium in basis points.
+     *      Cannot be higher than 30% (3000 basis points).
+     * @param _feePremiumBps Fee premium in basis points
+     */
+    function setFeePremiumBps(uint32 _feePremiumBps) external onlyGovernor {
+        _setFeePremiumBps(_feePremiumBps);
+    }
+
+    /**
+     * @dev Set the fee premium in basis points
+     * @param _feePremiumBps Fee premium in basis points
+     */
+    function _setFeePremiumBps(uint32 _feePremiumBps) internal {
+        require(_feePremiumBps <= 3000, "Fee premium too high"); // 30%
+
+        feePremiumBps = _feePremiumBps;
+        emit CCTPFeePremiumBpsSet(_feePremiumBps);
+    }
+
+    /***************************************
+             CCTP message handling
+    ****************************************/
+
+    /**
+     * @dev Handles a finalized CCTP message
+     * @param sourceDomain Source domain of the message
+     * @param sender Sender of the message
+     * @param finalityThresholdExecuted Fidelity threshold executed
+     * @param messageBody Message body
+     */
+    function handleReceiveFinalizedMessage(
+        uint32 sourceDomain,
+        bytes32 sender,
+        uint32 finalityThresholdExecuted,
+        bytes memory messageBody
+    ) external override onlyCCTPMessageTransmitter returns (bool) {
+        return
+            _handleReceivedMessage(
+                sourceDomain,
+                sender,
+                finalityThresholdExecuted,
+                messageBody
+            );
+    }
+
+    /**
+     * @dev Handles an unfinalized but safe CCTP message
+     * @param sourceDomain Source domain of the message
+     * @param sender Sender of the message
+     * @param finalityThresholdExecuted Fidelity threshold executed
+     * @param messageBody Message body
+     */
+    function handleReceiveUnfinalizedMessage(
+        uint32 sourceDomain,
+        bytes32 sender,
+        uint32 finalityThresholdExecuted,
+        bytes memory messageBody
+    ) external override onlyCCTPMessageTransmitter returns (bool) {
+        // Make sure the contract is configured to handle unfinalized messages
+        require(
+            minFinalityThreshold == 1000,
+            "Unfinalized messages are not supported"
+        );
```

Comment:
Quick question, what's the reason for only supporting fast transfer and not standard transfer?